asmcli/lib/installation_dependencies.sh

```diff
@@ -317,8 +325,6 @@ generate_membership_name() {
       RAND="$(tr -dc "a-z0-9" </dev/urandom | head -c8 || true)"
       MEMBERSHIP_NAME="${CLUSTER_NAME:0:54}-${RAND}"
     fi
-  else
-    MEMBERSHIP_NAME="$(date +%s%N)"
```

Comment:
I think removing this line means that anything off-GCP is going to need to be registered beforehand because we don't have a way of generating a membership name from nothing.

That's fine in principle, but I'd rather not add a potentially breaking change in case someone's relying on this. Can you think of a way to generate a more meaningful name from information that we're guaranteed to have even off-GCP?

(I don't really know how off-GCP clusters would get this far in the installation process without registering, but somehow it happens...)

Reply:
That makes sense, I was initially thinking it would be a requirement for the cluster to be registered (and without the `--enable-registration` or `--enable-all` flag, the script would fail). But I see that the registration uses this function to set parameters. Changing this back!